### PR TITLE
Move commands from solana cli #137

### PR DESCRIPTION
**Pull request by s-medvedev**
Move commands from solana-cli. All commands have the same output as in solana-cli.
**create-program-address**:
```
$ neon-cli create-program-address --help
neon-cli-create-program-address 
Generate a program address

USAGE:
    neon-cli create-program-address [FLAGS] [OPTIONS] <SEED_STRING>

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information
    -v, --verbose    Show additional information

OPTIONS:
    -C, --config <PATH>              Configuration file to use [default: /home/semen/.config/solana/cli/config.yml]
        --evm_loader <EVM_LOADER>    Pubkey for evm_loader contract
    -u, --url <URL>                  URL for Solana node [default: http://localhost:8899]

ARGS:
    <SEED_STRING>    The seeds (a string containing seeds in hex form, separated by spaces)
```

**create-ether-account**:
```
$ neon-cli create-ether-account --help
neon-cli-create-ether-account 
Create ethereum account

USAGE:
    neon-cli create-ether-account [FLAGS] [OPTIONS] <ether>

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information
    -v, --verbose    Show additional information

OPTIONS:
    -C, --config <PATH>              Configuration file to use [default: /home/semen/.config/solana/cli/config.yml]
        --evm_loader <EVM_LOADER>    Pubkey for evm_loader contract
    -u, --url <URL>                  URL for Solana node [default: http://localhost:8899]
        --lamports <lamports>         [default: 0]
        --space <space>              Length of data for new account [default: 0]

ARGS:
    <ether>    Ethereum address
```

**deploy**:
```
$ neon-cli deploy --help
neon-cli-deploy 
Deploy a program

USAGE:
    neon-cli deploy [FLAGS] [OPTIONS] <PROGRAM_FILEPATH>

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information
    -v, --verbose    Show additional information

OPTIONS:
    -C, --config <PATH>              Configuration file to use [default: /home/semen/.config/solana/cli/config.yml]
        --evm_loader <EVM_LOADER>    Pubkey for evm_loader contract
    -u, --url <URL>                  URL for Solana node [default: http://localhost:8899]

ARGS:
    <PROGRAM_FILEPATH>    /path/to/program.o
```

### CODE REVIEW

File: evm_loader/program/src/account_data.rs [hunk 1]
```diff
@@ -136,7 +136,7 @@ impl AccountData {
 }
 
 impl Account {
-    const SIZE: usize = 20+1+8+32+32+1+32;
+    pub const SIZE: usize = 20+1+8+32+32+1+32;
```
Comment (anton-lisanin) [inline evm_loader/program/src/account_data.rs:139]: Why you want to make it public? You don't use it.

**Review by sinev-valentine — DISMISSED** [review]

File: evm_loader/cli/src/main.rs [hunk 2]
```diff
@@ -66,6 +122,453 @@ fn emulate(solana_url: String, evm_loader: Pubkey, contract_id: H160, caller_id:
...
+    check_account_for_fee(
+        &config.rpc_client,
+        &config.signer.pubkey(),
+        &fee_calculator,
+        &finalize_message)?;
+
+    let mut finalize_tx = Transaction::new_unsigned(finalize_message);
+
+    finalize_tx.try_sign(&[&*config.signer], blockhash)?;
+    println!("signed: {:x?}", finalize_tx);
```
Comment (anton-lisanin) [inline evm_loader/cli/src/main.rs]: Are you sure you want to print logs into stdout? eprintln! is used anywhere else.

File: evm_loader/program/src/account_data.rs (hunk 1 above)
Comment (s-medvedev) [reply evm_loader/program/src/account_data.rs:139]: `SIZE` used in https://github.com/neonlabsorg/solana-program-library/pull/169/files#diff-2d7825462cfe18097d90eff351f21b30572b259b71f90a9fa5c1256f53da87c2R375

File: evm_loader/cli/src/main.rs [hunk 3]
```diff
@@ -182,35 +685,143 @@ fn main() {
...
+                    .unwrap_or(&cli_config.json_rpc_url),
+            );
+
+            let evm_loader = pubkey_of(&app_matches, "evm_loader")
+                    .unwrap_or_else(|| {
+                        eprintln!("Need specify evm_loader");
+                        exit(1);
+                    });
+
+            let verbose = app_matches.is_present("verbose");
```
Comment (anton-lisanin) [inline evm_loader/cli/src/main.rs]: verbose flag does nothing. May be remove it? Or use it and print detailed logs.

File: evm_loader/cli/src/main.rs [hunk 4]
```diff
@@ -66,6 +122,453 @@ fn emulate(solana_url: String, evm_loader: Pubkey, contract_id: H160, caller_id:
...
+) -> CommandResult {
+
+    let ACCOUNT_HEADER_SIZE = 1+Account::SIZE;
+    let CONTRACT_HEADER_SIZE = 1+Contract::SIZE;
+
+    let program_data = read_program_data(program_location)?;
+    let program_code_len = CONTRACT_HEADER_SIZE + program_data.len() + 2*1024;
+    let minimum_balance_program = config.rpc_client.get_minimum_balance_for_rent_exemption(ACCOUNT_HEADER_SIZE)?;
+    let minimum_balance_code = config.rpc_client.get_minimum_balance_for_rent_exemption(program_code_len)?;
+    let minimum_caller_balance = config.rpc_client.get_minimum_balance_for_rent_exemption(ACCOUNT_HEADER_SIZE)?;
```
Comment (anton-lisanin) [inline evm_loader/cli/src/main.rs]: minimum_balance_program and minimum_caller_balance will have the same value. Remove the duplicated RPC request.

Beside this, minimum_caller_balance is unused variable and can be completely removed.

Comment (s-medvedev) [reply evm_loader/cli/src/main.rs]: You are right. Fixed.

File: evm_loader/cli/src/main.rs (hunk 3 above)
Comment (s-medvedev) [reply evm_loader/cli/src/main.rs]: I added `log` module and used `verbose` flag to sets its level.

File: evm_loader/cli/src/main.rs (hunk 2 above)
Comment (s-medvedev) [reply evm_loader/cli/src/main.rs]: Fixed

File: evm_loader/cli/src/main.rs (hunk 3 above)
Comment (anton-lisanin) [reply evm_loader/cli/src/main.rs]: CLI imports the code from the evm_loader. How `log` module will interact with `debug_print!` from there?

**Review by anton-lisanin — DISMISSED** [review]

**Review by anton-lisanin — APPROVED** [review]

**Pull request merged**